### DISCUSSION TITLE
Reverse Mode: Read-Write

### DISCUSSION THREAD

**Discussion by TippyLion28** [rfjakob/gocryptfs#717] [Ideas & Feature requests]
Hi,

I am using SyncThing & gocryptfs in reverse mode to sync photos from my phone to a remote device. I want to be able to delete/move encrypted files at the remote end and have those changes propogated back to the phone. I am unable to do so at the moment because reverse mode does not yet support read-write.

This would be super helpful :))

**Comment by HarryLang** [comment]
I agree that this would be very helpful.  I would like to use rsync + gocryptfs.

It seems that one way to achieve this now would be to operate in regular mode, rather than reverse mode.  The drawback to this approach is that all local operations now have the overhead of being decrypted (for reads) and encrypted (for writes).  It would be ideal (for some use cases, like my own) to have this overhead only occur while syncing.

**Comment by axet** [comment]
The only reverse write mode available so far is by using old encfs 1.9.5.

**Comment by haarp** [comment]
Chiming in to say that writable reverse mode cipherdirs would be very useful indeed. In my use case, they would allow restoring encrypted backups done by [BackupPC](https://backuppc.github.io/backuppc/) directly.
